frontend/apps/filemanager/filemanagerbookinfo.lua

```diff
@@ -227,32 +275,117 @@ function BookInfo:onShowBookDescription(description, file)
+    local cover_bb
+    local custom_cover = self:getCustomBookCover(file or (doc and doc.file))
+    if not force_orig and custom_cover then
+        local img_widget = ImageWidget:new{
+            file = custom_cover,
+        }
+        cover_bb = img_widget:getImage()
+        img_widget:free()
+    else
+        local not_is_doc = not doc and true or false
+        if not_is_doc then
+            doc = DocumentRegistry:openDocument(file)
+            if doc and doc.loadDocument then -- CreDocument
+                doc:loadDocument(false) -- load only metadata
+            end
+        end
+        if doc then
+            cover_bb = doc:getCoverPageImage()
+            if not_is_doc then
+                doc:close()
+            end
+        end
+    end
+    return cover_bb
+end
+
+function BookInfo:getCustomBookCover(file)
+    local sidecar_dir = DocSettings:getSidecarDir(file, "doc")
+    local cover_file = DocSettings:getCoverFile(sidecar_dir)
+    if not cover_file then
+        sidecar_dir = DocSettings:getSidecarDir(file, "dir")
+        cover_file = DocSettings:getCoverFile(sidecar_dir)
+    end
+    return cover_file
+end
+
+function BookInfo:setCustomBookCover(file, book_props, caller_callback, cover_file)
+    local function kvp_update()
+        self.updated = true
+        self.kvp_widget:onClose()
+        if self.document then
+            self:onShowBookInfo()
         else
-            UIManager:show(InfoMessage:new{
-                text = _("No cover image available."),
-            })
+            self:show(file, book_props, caller_callback)
         end
-        document:close()
+    end
+    if cover_file then -- reset custom cover
+        local ConfirmBox = require("ui/widget/confirmbox")
+        local confirm_box = ConfirmBox:new{
+            text = _("Reset custom cover?"),
+            ok_text = _("Reset"),
+            ok_callback = function()
+                if os.remove(cover_file) then
+                    -- remove empty sdr folder
+                    local sidecar_dir = util.splitFilePathName(cover_file)
+                    if sidecar_dir == DocSettings:getSidecarDir(file, "doc") then
+                        os.remove(sidecar_dir)
+                    else
+                        util.removePath(sidecar_dir)
+                    end
+                    kvp_update()
+                end
+            end,
+        }
+        UIManager:show(confirm_box)
+    else -- choose an image and set custom cover
+        local PathChooser = require("ui/widget/pathchooser")
+        local path_chooser = PathChooser:new{
+            select_directory = false,
+            file_filter = function(filename)
+                return DocSettings.cover_ext[util.getFileNameSuffix(filename)]
+            end,
+            onConfirm = function(image_file)
+                local sidecar_dir
+                local sidecar_file = self:getCustomBookCover(file) -- existing cover file
+                if sidecar_file then
```

Comment:
This variable should probably have another name than `sidecar_file` ? sidecar_cover_file ?

Reply:
> sidecar_cover_file

Actually it is. Later I do `util.splitFilePathName(sidecar_file)` that should be applied to either `sidecar_cover_file` or `sidecar_file`, so using the same name makes the code shorter.
The comment explains the idea I hope.

Reply:
Well, ok :) (took me a few minutes to understand what you meant :).
So, as here we are not in DocSettings, where sidecar_file has an explicite meaning, we can read it here as "some sidecar file" :)

Reply:
> we can read it here as "some file in the sidecar dir"

Exactly!